dataset.py

```diff
@@ -1,77 +1,167 @@
+
+ 
 class Flickr8kDataset(Dataset):
+    """
+    Dataset for Flickr8k data treatment
+    """
+
+    def __init__(self, dataset_folder='./dataset',transform=None,reduce=False,max_size=5000):
+        super().__init__()
+        self.transform = transform
+        self.images_folder = os.path.join(dataset_folder,'Images')
+        self.dataframe = pd.read_csv(open(os.path.join(dataset_folder,'captions.txt'),'r'))
 
-	def __init__(self, dataset_folder='./dataset',transform=None):
-		super().__init__()
-		self.transform = transform
-		self.images_folder = os.path.join(dataset_folder,'images')
-		self.dataframe = pd.read_csv(open(os.path.join(dataset_folder,'captions.txt'),'r'))
+        self.caption = self.dataframe['caption']
+        self.vocab = Vocabulary()
+        self.vocab.build_vocabulary(self.caption.tolist(),reduce,max_size)
+        
 
-	def read_image(self,filename):
-		"""
-		Reads the image file based on the filename
+    def read_image(self,filename):
+        """
+        Reads the image file based on the filename
 
-		Args:
-			filename (string): Filename that contains the image
+        Args:
+            filename (string): Filename that contains the image
 
-		Returns:
-			[Image]: Returns a PIL.Image object 
-		"""
-		image_path = os.path.join(self.images_folder, filename)
-		image = Image.open(image_path)
-		return image
+        Returns:
+            [Image]: Returns a PIL.Image object 
+        """
+        image_path = os.path.join(self.images_folder, filename)
+        image = Image.open(image_path)
+        return image
 
-	def __len__(self):
-		return len(self.dataframe)
+    def __len__(self):
+        return len(self.dataframe)
 
-	def __getitem__(self,idx):
-		"""
-		Process the dataset according to our needs.
-		The image needs to be resize and normalize.
+    def __getitem__(self,idx):
+        """
+        Process the dataset according to our needs.
+        The image needs to be resize and normalize.
 
-		Args:
-			idx (int): Index of the dataset element to be returned
+        Args:
+            idx (int): Index of the dataset element to be returned
 
-		Returns:
-			torch.Tensor: 	Processed image
-			string:			Caption of the image
-		"""
-		filename, caption = self.dataframe.iloc[idx,:]
-		image = self.read_image(filename)
+        Returns:
+            torch.Tensor: 	Processed image
+            string:			Caption of the image
+        """
+        filename, caption = self.dataframe.iloc[idx,:]
+        image = self.read_image(filename)
 
-		if self.transform:
-			image = self.transform(image)
+        if self.transform:
+            image = self.transform(image)
 
-		return image, caption
+        tok_vec = []
+        tok_vec += [self.vocab.word_to_index['<START>']]
+        tok_vec += self.vocab.numericalize(caption)
+        tok_vec += [self.vocab.word_to_index['<END>']]
 
+        return image, torch.tensor(tok_vec)
+class CapsCollate:
```

Comment:
same for this one. Maybe a module should be created in the project that contains the whole dataset classes structure